lib/common.ts

```diff
@@ -25,14 +25,20 @@ export namespace Constants {
 }
 
 /** 
- * According to the RDF semantics, the correct representation of a dataset is a Set of Quads.
+ * According to the RDF semantics, the correct representation of a dataset is a Set of Quads. That is
+ * the structure used internally in the algorithm.
  */
 export type Quads = Set<rdf.Quad>;
 
+/**
+ * This is the external, "input" view of the dataset
+ */
+export type InputQuads = Iterable<rdf.Quad>;
```

Comment:
I would still be inclined to make this a `DatasetCore | TermSet` rather than just an `Iterable` as the former types will not have duplicate quads when iterate over; no such guarantee for the `Iterable` type.

If keeping the `Iterable` type then I would just add a note in your documentation that the input *must* not have repeated quads in the iteration in order for the algorithm to run correctly.

Reply:
Yes, I have it on my machine already, will push it with the other adjustments.